pkg/controller/utils/component.go

```diff
@@ -147,6 +149,16 @@ func (c componentHandler) CreateOrUpdate(ctx context.Context, component render.C
 		status.AddDeployments(deployments)
 		status.AddStatefulSets(statefulsets)
 	}
+
+	for _, obj := range objsToDelete {
+		err := c.client.Delete(ctx, obj)
+		if err != nil {
+			logCtx := ContextLoggerForResource(c.log, obj)
+			logCtx.Error(err, "Error deleting object %v", obj)
+			return err
+		}
```

Comment:
We'll need to update the status manager with the deleted resources, similar to this bit of code: https://github.com/tigera/operator/pull/386/files#diff-73950099f4736837cc6709a49bcb595aR147-R151

But instead using the `status.Remove` equivalent (or whatever it's called).